### PR TITLE
upgrade sbt-gsp, gsp-math, cats, scalatest

### PR DESCRIPTION
**Pull request by tpolecat**
This adds the `sbt-gsp` plugin and updates `gsp-math` and upstream dependencies including ScalaTest, which is the source of most of the changes here.

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -9,9 +9,12 @@ val scalaJSVersion =
 libraryDependencies ++= Seq(
   "org.postgresql" % "postgresql"  % "42.2.6", // needed by flyway
   "org.slf4j"      % "slf4j-nop"   % "1.7.28", // to silence some log messages
-  "org.typelevel" %% "cats-core"   % "1.6.1",
-  "org.typelevel" %% "cats-effect" % "0.8"
+  "org.typelevel" %% "cats-core"   % "2.0.0-RC1",
```
Comment (geminihlswadmin) [inline project/plugins.sbt]: Could this be the problem? Doesn't this add cats to the sbt classpath?

Comment (tpolecat) [reply project/plugins.sbt]: No, that's not a big deal. It was happening before I did that. Also you're logged in as @geminihlswadmin for some reason.

**Comment by tpolecat** [comment]
Ok the tests run now and everything passes. Thanks @cquiroz for spotting the problem in gsp-math. I will resolve conflicts and with luck this will be good to go.

**Comment by tpolecat** [comment]
Ok assuming this passes can you have a look @cquiroz since you seem to be around? Good time to merge this since nobody is around.

**Comment by cquiroz** [comment]
I looks fine in general. I'd like to give it a spin starting up the seqexec and do a few simulated runs
I'll try to do it today

**Comment by tpolecat** [comment]
:+1: sounds good, thanks

File: modules/ephemeris/src/test/scala/gem/horizons/HorizonsEphemerisQueryPagingPropSpec.scala [hunk 2]
```diff
@@ -83,7 +80,7 @@ class HorizonsEphemerisQueryPagingPropSpec extends PropSpec with PropertyChecks
     }
   }
 
-  property("each element is spaced exactly by duration") {
+  test("each element is spaced exactly by duration") {
```
Comment (cquiroz) [inline modules/ephemeris/src/test/scala/gem/horizons/HorizonsEphemerisQueryPagingPropSpec.scala]: I don't recall `property` was it related to scalacheck?

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/SeqexecApp.scala [hunk 3]
```diff
@@ -31,6 +29,9 @@ final class SeqexecLauncher[F[_]](implicit val F: Sync[F], L: LiftIO[F]) {
     setLoggerThreshold("", AllThreshold)
   }
 
+  private implicit val ioContextShift: ContextShift[IO] =
```
Comment (cquiroz) [inline modules/seqexec/web/client/src/main/scala/seqexec/web/client/SeqexecApp.scala]: Why didn't we need this before?

File: modules/ephemeris/src/test/scala/gem/horizons/HorizonsEphemerisQueryPagingPropSpec.scala (hunk 2 above)
Comment (tpolecat) [reply modules/ephemeris/src/test/scala/gem/horizons/HorizonsEphemerisQueryPagingPropSpec.scala]: Yeah you can write tests with straight scalacheck and no scalatest.

File: modules/seqexec/web/client/src/main/scala/seqexec/web/client/SeqexecApp.scala (hunk 3 above)
Comment (tpolecat) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/SeqexecApp.scala]: Good question, I will look more closely.

Comment (tpolecat) [reply modules/seqexec/web/client/src/main/scala/seqexec/web/client/SeqexecApp.scala]: `IO.fromFuture` now takes `ContextShift[IO]`. Daniel added late in the release cycle https://github.com/typelevel/cats-effect/commit/ebab1db7b9a116c939c60213339743d5b9907ac9

**Pull request merged**